eng/scripts/get-aspire-cli.sh

```diff
@@ -0,0 +1,437 @@
+                ;;
+            -q|--quality)
+                BUILD_QUALITY="$2"
+                shift 2
+                ;;
+            --os)
+                OS="$2"
+                shift 2
+                ;;
+            --architecture)
+                ARCH="$2"
+                shift 2
+                ;;
+            -k|--keep-archive)
+                KEEP_ARCHIVE=true
+                shift
+                ;;
+            -v|--verbose)
+                VERBOSE=true
+                shift
+                ;;
+            -h|--help)
+                SHOW_HELP=true
+                shift
+                ;;
+            *)
+                printf "Error: Unknown option '%s'\n" "$1" >&2
+                printf "Use --help for usage information.\n" >&2
+                exit 1
+                ;;
+        esac
+    done
+}
+
+# Function for verbose logging
+say_verbose() {
+    if [[ "$VERBOSE" == true ]]; then
+        printf "%s\n" "$1"
+    fi
+}
+
+# Function to detect OS
+get_os() {
+    local uname_s
+    uname_s=$(uname -s)
+
+    case "$uname_s" in
+        Darwin*)
+            printf "osx"
+            ;;
+        Linux*)
+            # Check if it's musl-based (Alpine, etc.)
+            if ldd --version 2>&1 | grep -q musl; then
+                printf "linux-musl"
+            else
+                printf "linux"
+            fi
+            ;;
+        CYGWIN*|MINGW*|MSYS*)
+            printf "win"
+            ;;
+        *)
+            printf "unsupported"
+            return 1
+            ;;
+    esac
+}
+
+# Function to detect architecture
+get_arch() {
+    local uname_m
+    uname_m=$(uname -m)
+
+    case "$uname_m" in
+        x86_64|amd64)
+            printf "x64"
+            ;;
+        aarch64|arm64)
+            printf "arm64"
+            ;;
+        i386|i686)
+            printf "x86"
+            ;;
+        *)
+            printf "unsupported"
+            return 1
+            ;;
+    esac
+}
+
+# Common function for HTTP requests with centralized configuration
+secure_curl() {
+    local url="$1"
+    local output_file="$2"
+    local timeout="${3:-300}"
+    local user_agent="${4:-$USER_AGENT}"
+    local max_retries="${5:-5}"
+
+    # FIXME: --cert-status is failing with `curl: (91) No OCSP response received`
+    curl \
+        --fail \
+        --silent \
+        --show-error \
+        --location \
+        --tlsv1.2 \
+        --max-time "$timeout" \
+        --user-agent "$user_agent" \
+        --max-redirs 10 \
+        --retry "$max_retries" \
+        --retry-delay 1 \
+        --retry-max-time 60 \
+        --output "$output_file" \
+        "$url"
+}
+
+# Download the Aspire CLI for the current platform
+download_aspire_cli() {
+    local url="$1"
+    local filename="$2"
+    local temp_filename="${filename}.tmp"
+
+    printf "Downloading from: %s\n" "$url"
+
+    # Use temporary file and move on success to avoid partial downloads
+    if secure_curl "$url" "$temp_filename" 300; then
```

Comment:
Is 300 going to be enough on slow internet connections?

Reply:
What number would you suggest? 10mins?

Reply:
I'm not sure. 10 min should be plenty, I think